### DISCUSSION TITLE
how to speed up scheduling tasks?

### DISCUSSION THREAD

**Discussion by tuwei1314** [argoproj/argo-workflows#12766] [Q&A] [answered]
It takes only a few seconds to run a job in the k8s cluster. It takes 10 seconds to arrange the same job in argo-workflow, and almost 10 seconds is spent on workflow scheduling. Is there any way to speed up the running of a job

**Comment by shuangkun** [comment]
Are there many workflows in the cluster? You can adjust the qps and burst of workflow-controller.

**Comment by tuwei1314** [comment]
just one workflows for test,when i test two task and use the dag,it will cost 20s but the pod just cost for 6s



---Original---
From: "shuangkun ***@***.***&gt;
Date: Sat, Mar 9, 2024 10:41 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [argoproj/argo-workflows] how to speed up the job of theargo-workflow (Discussion #12766)




 
Are there many workflows in the cluster? You can adjust the qps and burst of workflow-controller.

**Comment by tuwei1314** [comment]
I personally understand that most of the time is spent on workflow scheduling, and it takes about 10 seconds to complete each task until the next one starts. Is there any optimization that can speed up scheduling




剑指龙门
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "剑指龙门"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年3月9日(星期六) 中午11:23
***@***.******@***.***&gt;;
***@***.***&gt;;
主题:&nbsp;Re: [argoproj/argo-workflows] how to speed up the job of theargo-workflow (Discussion #12766)



just&nbsp; one&nbsp; workflows&nbsp; for&nbsp; test,when&nbsp; i test&nbsp; two&nbsp; task and use the&nbsp; dag,it will&nbsp; cost&nbsp; 20s but the&nbsp; pod&nbsp; just&nbsp; cost&nbsp; for&nbsp; 6s



---Original---
From: "shuangkun ***@***.***&gt;
Date: Sat, Mar 9, 2024 10:41 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [argoproj/argo-workflows] how to speed up the job of theargo-workflow (Discussion #12766)




 
Are there many workflows in the cluster? You can adjust the qps and burst of workflow-controller.

**Comment by shuangkun** [comment]
If there are a large number of workflows in the cluster, slow scheduling is normal. If there is only a simple workflow in the cluster, scheduling is very fast. If there is a sample workflow, we can take a look at it together.

**Comment by agilgur5** [comment, marked as answer]
This sounds like a duplicate of #11275. The Controller [environment variable](https://argo-workflows.readthedocs.io/en/latest/environment-variables/#controller) `DEFAULT_REQUEUE_TIME` controls the workflow queue rate limiter. You can decrease that, but that in turn can [overwhelm your k8s API server](https://argo-workflows.readthedocs.io/en/latest/running-at-massive-scale/#overwhelmed-kubernetes-api). It currently defaults to `10s`.

You can also speed up Pod start-up time by caching images, having available resources (i.e. over-provisioning and not needing to auto-scale as much), and more. All of those are conventional k8s tuning knobs, not specific to Argo.

#7144 also discusses the use-case of re-using Pods between tasks (which is not appropriate for all users).
